pkg/operator/collection.go

```diff
@@ -138,9 +140,14 @@ func (r *collectionReconciler) Reconcile(ctx context.Context, req reconcile.Requ
 	if err := r.ensureCollectorConfig(ctx, &config.Collection); err != nil {
 		return reconcile.Result{}, errors.Wrap(err, "ensure collector config")
 	}
-	if err := r.updateCRDStatus(ctx); err != nil {
-		return reconcile.Result{}, errors.Wrap(err, "update crd status")
+
+	// Reconcile any status updates.
+	for _, obj := range r.statusUpdates {
+		if err := r.client.Status().Update(ctx, obj); err != nil {
```

Comment:
Mh, maybe log an error and try the next one regardless?

Also `statusUpdates` doesn't seem to ever be reset.

Reply:
Ah good call - done.